### PR TITLE
Add Clear Playlist Button

### PR DESCRIPTION
**Pull request by chamow97**
- Clear the playlist array.
- Reset Total Duration.
- Reload the Page.
- Set Title of the page to "Youtube Fast Playlist"

### CODE REVIEW

File: js/player.js [hunk 1]
```diff
@@ -150,6 +151,24 @@ function($, playlist, ytIframe){
             }
         }
     }
+    var clearPlaylist = function(){
```
Comment (udiboy1209) [inline js/player.js]: You should keep a newline between two functions.

File: js/playlist.js [hunk 2]
```diff
@@ -187,6 +187,9 @@ function(ytData,ytIframe,template,$) {
         savePlaylist();
     }
 
+
+
+
```
Comment (udiboy1209) [inline js/playlist.js]: Unnecessary change? please remove this. It is always good to check your changes using `git diff` before pushing!

File: js/player.js [hunk 3]
```diff
@@ -150,6 +151,24 @@ function($, playlist, ytIframe){
...
+      //save the playlist
+      localStorage.setItem('playlist', JSON.stringify(playlist));
+      localStorage.setItem('new_playlist_saved', JSON.stringify(true));
+      //reset duration
+      localStorage.setItem('total_duration', JSON.stringify(totalDuration));
+      //reload page
+      location.reload();
+      //change title
+      document.title= "Youtube Fast Playlist";
+    }
```
Comment (udiboy1209) [inline js/player.js]: This whole function doesn't make sense here, because it makes changes to the playlist. It should be placed in `playlist.js`. And why is `location.reload()` necessary?

File: js/player.js [hunk 4]
```diff
@@ -16,6 +16,7 @@ function($, playlist, ytIframe){
 
         $('#play_prev').on('click', playPrev);
         $('#play_next').on('click', playNext);
+        $('#clear_playlist').on('click', clearPlaylist);
```
Comment (udiboy1209) [inline js/player.js]: When you move the `clearPlaylist` function to playlist.js, you can access it using `playlist.clearPlaylist`

File: index.html [hunk 5]
```diff
@@ -69,6 +70,10 @@ <h5 id="totalDuration">00:00:00</h5>
                         <div id="share" class="col s3 center waves-effect btn-play-control">
                             <i class="material-icons small" title="Copy Share Link">share</i>
                         </div>
+                        <div id="clear_playlist" class="col s2 center waves-effect btn-play-control">
+                            <i class="material-icons small" title="Clear Playlist">clear</i>
+                        </div>
```
Comment (udiboy1209) [inline index.html]: The different sized columns look weird currently. I have a suggestion, why don't you create an extended options menu which will dropdown (using [more_vert icon](https://material.io/icons/#ic_more_vert) ) and show both "Share" and "Clear playlist". That will look less cluttered.

You can copy the dropdown implementation of `play_mode`.

Comment (chamow97) [reply index.html]: Okay. I will create a extended icon

File: js/playlist.js (hunk 2 above)
Comment (chamow97) [reply js/playlist.js]: Sorry for that.

File: js/player.js (hunk 3 above)
Comment (chamow97) [reply js/player.js]: Thought reloading page will make it more cleaner. I will look on how to do the same without reloading the page.

Comment (udiboy1209) [reply js/player.js]: Another suggestion is that, when you set `playlist = []` and `total_duration = 0`, calling `savePlaylist()`  and  `saveDuration()` will be cleaner instead of directly setting it in `localStorage`.

Comment (chamow97) [reply js/player.js]: Okay.

File: js/playlist.js [hunk 6]
```diff
@@ -187,6 +187,21 @@ function(ytData,ytIframe,template,$) {
...
+      $('#playlist li').remove();
+      //reset playlis duration
+      $('#totalDuration').text("00:00:00");
+      //save the playlist
+      savePlaylist();
+      //reset duration
+      saveDuration();
+      //change title
+      document.title= "Youtube Fast Playlist";
+    }
```
Comment (udiboy1209) [inline js/playlist.js]: 3 things:

 - Rest of the code uses 4 space indents, please maintain that
 - You don't need to comment every line! The function `savePlaylist()` is descriptive enough in itself, so you don't need the extra comment. Ideally you should simply explain what the function does at the top of the function declare, and if it's very unclear add hints in the function body.
 - Everything works great right now, except that the playing video doesn't stop. Add `ytIframe.player.stopVideo();` to stop the video too.

File: index.html [hunk 7]
```diff
@@ -66,8 +67,16 @@ <h5 id="totalDuration">00:00:00</h5>
                                 <li id="repeat_shuffle"><i class="material-icons small" title="Shuffle">shuffle</i></li>
                             </ul>
                         </div>
-                        <div id="share" class="col s3 center waves-effect btn-play-control">
-                            <i class="material-icons small" title="Copy Share Link">share</i>
+                        <div class="col s3 center dropdown-button btn-flat" data-activates="more_options">
+                            <i id="more" class="material-icons small" title="More Options">expand_more</i>
```
Comment (udiboy1209) [inline index.html]: Using more_vert instead of expand_more would look better I guess. I think expand_more is used to expand visible sections to full and not for extra options.

**Comment by udiboy1209** [comment]
@chamow97 great work! :tada:
